src/core/ext/filters/client_channel/lb_policy/priority/priority.cc

```diff
@@ -501,6 +518,136 @@ void PriorityLb::SelectPriorityLocked(uint32_t priority) {
                                         child->GetPicker());
 }
 
+//
+// PriorityLb::ChildPriority::DeactivationTimer
+//
+
+PriorityLb::ChildPriority::DeactivationTimer::DeactivationTimer(
+    RefCountedPtr<PriorityLb::ChildPriority> child_priority)
+    : child_priority_(std::move(child_priority)) {
+  if (GRPC_TRACE_FLAG_ENABLED(grpc_lb_priority_trace)) {
+    gpr_log(GPR_INFO,
+            "[priority_lb %p] child %s (%p): deactivating -- will remove in "
+            "%" PRId64 "ms",
+            child_priority_->priority_policy_.get(),
+            child_priority_->name_.c_str(), child_priority_.get(),
+            kChildRetentionInterval.millis());
+  }
+  GRPC_CLOSURE_INIT(&on_timer_, OnTimer, this, nullptr);
+  Ref(DEBUG_LOCATION, "Timer").release();
```

Comment:
Seeing that the ExecCtx::Now API has the time cache optimization on by default, I'm ok with this reasoning.

That said, the time cache has turned out to cause subtle and hard-to-debug issues more than once before. So IMO our policy should be to invalidate the time before every timer, and not invalidate it only if it's determined to be needed for performance (until that PR is merged).

But, granted there's no indication that this particular usage would cause issues.

Reply:
I think we have consensus that we want to get to a future where we do time caching on the data plane but not anywhere on the control plane.  However, we need to find a reasonable API for doing that; it is not sustainable to expect all timer callers to explicitly call `InvalidateNow()`, and I would be opposed to that as a general approach to addressing this problem.  I think we are likely to find a better approach as part of the promise conversion work that Craig is doing.

In any case, this discussion is far outside of the scope of this PR.